### PR TITLE
Addin Hippopt Planner 

### PR DESCRIPTION
**Pull request by CarlottaSartore**
Depends on https://github.com/ami-iit/adam/pull/74

Addining planner based on Hippopt library https://github.com/ami-iit/hippopt/tree/main

### CODE REVIEW

File: examples/hippopt_walking.ipynb [hunk 1]
```diff
@@ -0,0 +1,179 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hippopt planner\n",
+    "This example, load a basic robot model (i.e. composed only of basic shapes), modifies the links of such a robot model by elongating the legs, plans a forward walking trajecotry using Hippopt walking planner  "
```
Comment (S-Dafarra) [inline examples/hippopt_walking.ipynb:8]:
```suggestion
    "This example loads a basic robot model (i.e. composed only of basic shapes), modifies the links of such a robot model by elongating the legs, plans a forward walking trajectory using Hippopt walking planner  "
```

File: src/comodo/robotModel/robotModel.py [hunk 2]
```diff
@@ -121,13 +125,16 @@ def compute_desired_position_walking(self):
...
+        for index, joint_name in enumerate(self.joint_name_list):
+            if "knee" in joint_name:
+                self.solver.subject_to(self.s[index] == desired_knee)
+            if "shoulder_roll" in joint_name: 
+                self.solver.subject_to(self.s[index] == shoulder_roll)
+            if "elbow" in joint_name: 
+                self.solver.subject_to(self.s[index] == elbow)    
+        
+        # self.solver.subject_to(self.s[9] == self.s[15])
+        # self.solver.subject_to(cs.norm_2(self.quat_pose_b[:4]) == 1.0)
```
Comment (S-Dafarra) [inline src/comodo/robotModel/robotModel.py:137]: Do you still need these lines?

File: src/comodo/robotModel/robotModel.py [hunk 3]
```diff
@@ -121,13 +125,16 @@ def compute_desired_position_walking(self):
...
-        self.solver.subject_to(self.s[1] == shoulder_roll)
-        self.solver.subject_to(self.s[5] == shoulder_roll)
-        self.solver.subject_to(self.s[9] == self.s[15])
+        for index, joint_name in enumerate(self.joint_name_list):
+            if "knee" in joint_name:
+                self.solver.subject_to(self.s[index] == desired_knee)
+            if "shoulder_roll" in joint_name: 
+                self.solver.subject_to(self.s[index] == shoulder_roll)
+            if "elbow" in joint_name: 
+                self.solver.subject_to(self.s[index] == elbow)
```
Comment (S-Dafarra) [inline src/comodo/robotModel/robotModel.py:134]: General comment, the class name is a generic "robotModel", but these lines are very specific to the ergoCub/iCub robots

File: src/comodo/robotModel/robotModel.py (hunk 2 above)
Comment (CarlottaSartore) [reply src/comodo/robotModel/robotModel.py]: Removed thanks !

File: src/comodo/robotModel/robotModel.py (hunk 3 above)
Comment (CarlottaSartore) [reply src/comodo/robotModel/robotModel.py:134]: Yes I agree, I think once we have merged the major PRs, we should work on a generalization of the robot model. I would do it in a second step

**Comment by CarlottaSartore** [comment]
@S-Dafarra  Sorry for the delay, I should have implemented/ answer all the comments

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**